### PR TITLE
fix `document.all property doesn't overridden` (close #1046)

### LINKED ISSUE #1046: document.all property doesn't overridden
The issue is reproducing in playground.
https://www.screencast.com/t/KwMlQHC6NS

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: src/client/sandbox/native-methods.js [hunk 1]
```diff
@@ -39,10 +41,13 @@ class NativeMethods {
         this.documentCreateTouchList     = doc.createTouchList || docProto.createTouchList;
 
         // getters/setters
-        const docPrototype = win.Document.prototype;
+        const docPrototype         = win.Document.prototype;
```
Comment (miherlosev) [inline src/client/sandbox/native-methods.js]: Let's move getters and setters related code to the separate method

File: src/client/sandbox/native-methods.js [hunk 2]
```diff
@@ -39,10 +39,13 @@ class NativeMethods {
...
+        const htmlDocPrototype     = win.HTMLDocument.prototype;
+        const documentAllPropOwner = docPrototype.hasOwnProperty('all') ? docPrototype : htmlDocPrototype;
```
Comment (miherlosev) [inline src/client/sandbox/native-methods.js]: Move to `utils/feature-detection.js` as `featureDetection.getDocumentAllPropertyOwner` function

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/native-methods.js (hunk 2 above)
Comment (LavrovArtem) [reply src/client/sandbox/native-methods.js]: We cannot import the `feature-detection` module into the `nativeMethods` because it is circular dependency.

Comment (miherlosev) [reply src/client/sandbox/native-methods.js]: Sorry. ‘NativeMethods.getDocumentAllPropertyOwner’

**Pull request merged**